packages/openapi-code-generator/src/typescript/common/type-builder.ts

```diff
@@ -126,11 +130,13 @@ export class TypeBuilder {
 
           // TODO better support
           const additionalProperties =
-            schemaObject.additionalProperties || properties.length === 0
+            schemaObject.additionalProperties
               ? "[key: string]: unknown"
               : ""
 
-          result.push(object([...properties, additionalProperties]))
+          const emptyObject = !additionalProperties && properties.length === 0 ? "[key: string]:  never" : ""
```

Comment:
Github spec includes schemas like:
```
              schema:
                type: object
                properties: {}
                additionalProperties: false
```
Often to represent an empty response. This should now type them more appropriately, hopefully without any majority usability issues introduced.